src/python/pants/util/contextutil.py

```diff
@@ -123,7 +123,7 @@ def _stdio_stream_as(src_fd, dst_fd, dst_sys_attribute, mode):
 
 
 @contextmanager
-def stdio_as(stdout_fd, stderr_fd, stdin_fd):
+def stdio_as(stdout_fd, stderr_fd, stdin_fd, binary_mode=False):
```

Comment:
afaict, the `binary_mode=True` case here is never exercised? should there be some `if PY3` conditional logic somewhere consuming this for when pants is executed with a 3.x interpreter?

Reply:
In the use cases so far (i.e. util tests), I haven't had to turn on `binary_mode=True`. We in general will want to use text mode rather than binary_mode, because this is how std streams work in Python 3 by default. 

Further, typically in Py3, if you do need to write bytes, you can access bytes through `stdin.buffer`.

The flag is only to give us the option if in porting everything else we find some use case where we want to write bytes directly to `stdin` etc, rather than unicode. For example, if we override `stdin` with a file that doesn't have this `buffer` functionality, and we want to write bytes, we'll have to trigger `binary_mode=True`.